src/main/java/org/kiwiproject/jersey/client/RegistryAwareClient.java

```diff
@@ -74,7 +74,7 @@ public WebTarget targetForService(ServiceIdentifier identifier) {
 
         var uri = registryClient.findServiceInstanceBy(instanceQuery)
                 .map(instance -> buildInstanceUri(identifier, instance))
-                .orElseThrow(() -> newMissingServiceRuntimeException(identifier));
+                .orElseThrow(() -> from(identifier));
```

Comment:
Generally I don't like to static import small factory methods (e.g. `of`, `from`, etc) since you lose context...